src/Cassandra/Mapping/TypeConversion/TypeConverter.cs

```diff
@@ -223,6 +224,29 @@ internal Delegate GetToDbConverter(Type pocoType, Type dbType)
 
             Type dbType = typeof (TDatabase);
             Type pocoType = typeof (TPoco);
+            
+            if (typeof(TPoco) == typeof(TDatabase))
+            {
+                Func<TPoco, TPoco> func = d => d;
+                return func;
+            }
```

Comment:
I was thinking that `ConvertToUdtFieldFromDbValue` didn't check that but the caller of `ConvertToUdtFieldFromDbValue` does so yeah I'll remove this. 

I don't understand what you mean by this though:
> and returns null when typeof(TPoco) == typeof(TDatabase) (that should occur)

Reply:
sorry, meant "that should not occur"

Reply:
I remove this check but I added one to an inner block for nullable types where the same method is called. I also added the check to the udt convert method just to ensure that if this method is ever used somewhere else this check still happens

Reply:
Actually this just broke a lot of tests because of the method call to itself when dealing with nullables. I've added the check again but removed it from `TryGetFromDbConverter` since it was redundant.